subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppCachingIntegrationTest.groovy

```diff
@@ -51,29 +52,116 @@ class CppCachingIntegrationTest extends AbstractCppInstalledToolChainIntegration
     }
 
     def 'compilation can be cached'() {
-        setupProjectInDirectory(temporaryFolder.testDirectory)
+        def f = new NativeCachingFixture(buildType: buildType)
+        setupProject()
 
         when:
-        withBuildCache().succeeds compilationTask
+        withBuildCache().run f.compilationTask
 
         then:
-        compileIsNotCached()
+        f.compileIsNotCached()
 
         when:
-        withBuildCache().succeeds 'clean', ':installDebug'
+        withBuildCache().run 'clean', installTask(buildType)
 
         then:
-        compileIsCached()
-        installation("build/install/main/debug").exec().out == app.expectedOutput
+        f.compileIsCached()
+        installation("build/install/main/${buildType.toLowerCase()}").exec().out == app.expectedOutput
+
+        where:
+        buildType << [debug, release]
+    }
+
+    def "compilation task is relocatable"() {
+        def f = new NativeCachingFixture(buildType: buildType)
+        def originalLocation = file('original-location')
+        def newLocation = file('new-location')
+        setupProject(originalLocation)
+        setupProject(newLocation)
+
+        when:
+        inDirectory(originalLocation)
+        withBuildCache().run f.compilationTask
+
+        def snapshotsInOriginalLocation = f.snapshotObjects(originalLocation)
+
+        then:
+        executedAndNotSkipped f.compilationTask
```

Comment:
Why not call `f.compilationIsNotCached()` here?

Reply:
changed.